frontend/src/app/pages/exercises/exercise/shared/exercise-map/utility/ol-geometry-helpers.ts

```diff
@@ -0,0 +1,124 @@
+import { Position } from 'digital-fuesim-manv-shared';
+import type { Size, UUID } from 'digital-fuesim-manv-shared';
+import { Feature } from 'ol';
+import type { Coordinate } from 'ol/coordinate';
+import type { Geometry } from 'ol/geom';
+import { Point, Polygon } from 'ol/geom';
+import type { WithPosition } from '../../utility/types/with-position';
+
+export interface PositionableElement {
+    readonly id: UUID;
+    readonly position: Position;
+}
+
+export type ResizableElement = PositionableElement & {
+    size: Size;
+};
+
+export type GeometryWithCoordinates = Geometry & {
+    getCoordinates: () => unknown;
+    setCoordinates: (coordinates: any[]) => void;
+};
+/**
+ * Typescript doesn't error when T doesn't satisfy
+ * getCoordinates: () => unknown.
+ * Instead, the inferred type is null. We exclude this type
+ * to exchange it with never
+ */
+export type Coordinates<T extends GeometryWithCoordinates> = Exclude<
+    ReturnType<T['getCoordinates']>,
+    null
+>;
+
+type ArrayElement<ArrayType> = ArrayType extends readonly (infer ElementType)[]
+    ? ElementType
+    : never;
+
+type SubstituteCoordinateForPoint<T> = T extends Coordinate
+    ? Position
+    : T extends Array<ArrayElement<T>>
+    ? SubstituteCoordinateForPoint<ArrayElement<T>>[]
+    : never;
+
+export type Positions<T extends GeometryWithCoordinates> =
+    SubstituteCoordinateForPoint<Coordinates<T>>;
+
+export interface CoordinatePair<T extends GeometryWithCoordinates> {
+    startPosition: Coordinates<T>;
+    endPosition: Coordinates<T>;
+}
+
+export const createPoint = (element: WithPosition<any>): Feature<Point> =>
+    new Feature(new Point(getCoordinatesPositionableElement(element)));
+
+export const createPolygon = (element: ResizableElement): Feature<Polygon> =>
+    new Feature(new Polygon([getCoordinateResizeableElement(element)]));
```

Comment:
What about refactoring this to object literals (or classes)?

So you would have a `pointGeometryHelper` and `polygonGeometryHelper` literal that implement an interface `GeometryHelper<Geometry>` specifying a `createGeometry`, `getCoordinates` and `interpolate` function.

You would then only have to pass this one helper object from the `FeatureManagers` to the parent class and e.g., the `MovableElementHelper` instead of three separate functions.

Reply:
You're right :)